src/components/OperationPanes/SetTrustLineFlags.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import OptionsTablePair from '../OptionsTable/Pair';
+import AssetPicker from '../FormComponents/AssetPicker.js';
+import PubKeyPicker from '../FormComponents/PubKeyPicker';
+import FlagfieldPicker from '../FormComponents/FlagfieldPicker';
+import CheckboxButtonPicker from "../FormComponents/CheckboxButtonPicker";
+import HelpMark from '../HelpMark';
+
+const flagsSet = {
+  'authorized': 'Authorized',
+  'authorizedToMaintainLiabilities': 'Authorized to maintain liabilites',
+};
+
+const flagsClear = {
+  'authorized': 'Authorized',
+  'authorizedToMaintainLiabilities': 'Authorized to maintain liabilites',
+  'clawbackEnabled': 'Clawback enabled',
+};
+
+export default function SetTrustLineFlags(props) {
+  return [
+    <OptionsTablePair label="Asset" key="asset">
+      <AssetPicker
+        value={props.values['asset']}
+        disableNative={true}
+        onUpdate={(value) => {props.onUpdate('asset', value)}}
+        />
+    </OptionsTablePair>,
+    <OptionsTablePair label="Trustor" key="trustor">
+      <PubKeyPicker
+        value={props.values['trustor']}
+        onUpdate={(value) => {props.onUpdate('trustor', value)}}
+        />
+    </OptionsTablePair>,
+    <OptionsTablePair
+      label={<span>Set Flags</span>}
+      optional={true} key="setFlags">
+      <CheckboxButtonPicker
+        className=""
+        value={props.values['setFlags'] || []}
+        onUpdate={(value) => {props.onUpdate('setFlags', value)}}
+        items={flagsSet}
+        />
+    </OptionsTablePair>,
```

Comment:
Is there a reason why you present the `setFlags` and `clearFlags` to the user directly? I only ask because I tried simplifying the API by letting people just set booleans ([brief discussion](https://github.com/stellar/js-stellar-base/pull/397#discussion_r598907531) for context if interested), so it's odd to see that complexity creep back in (alongside the `reduce(...)` code below to map *back* to the simplification).

Reply:
The "final" list is a combination of both (reduce just merges the two). I split them on the UI to make it obvious which flags are to be set and which to be cleared (`clawbackEnabled` can only be cleared). Are you suggesting that we should have a single list and by default all of them are cleared? Sorry if I don't get it.

`flags` submitted to the API are like this:

```javascript
}  
  authorized: true,
  authorizedToMaintainLiabilities: false,
  trustlineClawback: true,
}
```

Reply:
Ah, I see what you're saying. I'm more asking why (or if?) you're submitting all of the flags regardless of whether or not the user interacted with them. Do you retrieve state first to set the flags accordingly? I wouldn't expect someone to want to set &/or clear all of the flags at once. 

To clarify, the API is as follows:

```js
sdk.Operation.setTrustLineFlags({
  /* other stuff */
  flags: {
    authorized: true, /* authorization will be set */
    clawbackEnabled: false, /* clawback will be cleared */
  }
}
```

With the above call, `authorizedToMaintainLiabilities` will **neither be set *nor* cleared**, meaning its state will be unchanged for the account. However, it appears that via the UI (correct me if I'm wrong) the flags are cleared by default, and all of the flags will always be either set &/or cleared.

Reply:
Okay, turns out my eyes don't work well :eyes:. For posterity: it appears only flags that are clicked on in the UI will be submitted to the operation, alleviating my concern!